File: web/src/types/customer.ts
import type { Row } from "./row";

export type { RowStatus } from "./row";

export interface CustomerData {
    firstName: string;
    lastName: string;
    nationalNumber: string;
    gender: string;
    customerType: 1 | 2;
    nationality: 1 | 2;
    age: number;
    bankBranch: string;
    addressId: number | null;
    province: string;
    district: string;
    openAddress: string;
    phoneId: number | null;
    phoneType: string;
    countryCode: string;
    areaCode: string;
    phoneNumber: string;
}

export type CustomerRow = Row<CustomerData>;

File: src/ziraat.Api/Models/Customer.cs
namespace ziraat.Api.Models;

public class Customer
{
    public int Id { get; set; }
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public string NationalNumber { get; set; } = string.Empty;
    public char Gender { get; set; }
    public byte CustomerType { get; set; }
    public byte Nationality { get; set; }
    public int Age { get; set; }
    public string BankBranch { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
}
